tests/automation.js

```diff
@@ -546,6 +546,31 @@ casper.test.begin("unit tests", 25 + gfxTests.length, function(test) {
         test.assertTextDoesntExist("FAIL");
     });
 
+    // The following test tests that a MIDlet can be started, paused and then started again.
+    casper
+    .thenOpen("http://localhost:8000/index.html?midletClassName=tests.background.BackgroundMIDlet1&jad=tests/midlets/background/pause.jad&jars=tests/tests.jar&logConsole=web,page&logLevel=log")
+    .withFrame(0, function() {
+        casper.waitForText("startApp", function() {
+            casper.waitForText("pauseApp", function() {
+                casper.waitForText("startApp", function() {
+                    var content = this.getPageContent();
+                    test.assertEquals(content.match(/startApp/g).length, 2, "MIDlet started twice");
```

Comment:
This happens to work, but I think it's only because the second *startApp* races the *waitForText* timeout. According to [the waitForText docs](http://casperjs.readthedocs.org/en/latest/modules/casper.html#waitfortext), it simply "Waits until the passed text is present in the page contents," which means the second `casper.waitForText("startApp", …)` call will find the first instance of *startApp* if the second hasn't appeared yet.

Fortunately, the test will still fail if the app doesn't get started twice; but it could also fail when it does, leading to a false negative. To ensure that doesn't happen, remove the *startApp* content before restarting the app or make the app write a different message to the content node the second time around.